src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1773,6 +1773,32 @@ static void UpdateCurrentState(EvmState state, int pc, long gas, int stackHead)
                             _txTracer.SetOperationTransientStorage(storageCell.Address, storageIndex, newValue, currentValue);
                         }
 
+                        break;
+                    }
+                case Instruction.MCOPY:
+                    {
+                        if (!spec.MCopyIncluded)
+                        {
+                            goto InvalidInstruction;
+                        }
+
+                        Metrics.MCopyOpcode++;
+
+                        stack.PopUInt256(out UInt256 dstPosition);
+                        stack.PopUInt256(out UInt256 srcPosition);
+                        stack.PopUInt256(out UInt256 length);
+
+                        long baseGasCost = GasCostOf.VeryLow * (((long)length + 31) / 32);
+                        if (!UpdateGas(baseGasCost, ref gasAvailable)) goto OutOfGas;
+
+                        if (!UpdateMemoryCost(vmState, ref gasAvailable, in srcPosition, length)) goto OutOfGas;
+                        Span<byte> loadedData = vmState.Memory.LoadSpan(in srcPosition, length);
+                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange(srcPosition, loadedData);
```

Comment:
Memory is changing on load?

Reply:
I was thinking of removing these and keeping on the loading line, and then saw that MLOAD does them as well, so I kept them